core/runtime/wavm/impl/intrinsic_functions.cpp

```diff
@@ -0,0 +1,634 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/wavm/impl/intrinsic_functions.hpp"
+
+#include "runtime/wavm/intrinsic_resolver.hpp"
+
+namespace kagome::runtime::wavm {
+
+  log::Logger logger;
+
+  WAVM::Intrinsics::Module *getIntrinsicModule_env() {
+    static WAVM::Intrinsics::Module module;
+    return &module;
+  }
+
+  std::stack<std::shared_ptr<host_api::HostApi>>
+      global_host_apis;  //< TODO(Harrm) fix this wild crutch
+
+  void pushHostApi(std::shared_ptr<host_api::HostApi> api) {
+    global_host_apis.emplace(std::move(api));
+  }
+
+  void popHostApi() {
+    global_host_apis.pop();
+  }
+
+  std::shared_ptr<host_api::HostApi> peekHostApi() {
+    return global_host_apis.top();
+  }
+
+  const WAVM::IR::MemoryType kIntrinsicMemoryType{
+      WAVM::IR::MemoryType(false, WAVM::IR::IndexType::i32, {20, UINT64_MAX})};
+
+  // actually used, because added to the intrinsic module
+  [[maybe_unused]] WAVM::Intrinsics::Memory env_memory{
+      getIntrinsicModule_env(),
+      kIntrinsicMemoryName.data(),
+      kIntrinsicMemoryType};
+
+  void registerHostApiMethods(IntrinsicResolver &resolver,
+                              const host_api::HostApi &host_api) {
+#define GENERATE_HOST_INTRINSIC(Ret, name, ...) \
+  resolver.addIntrinsic(#name, &kagome::runtime::wavm::name##Intrinsic)
+
+#define GENERATE_STUB_INTRINSIC(Ret, name, ...) \
+  resolver.addIntrinsic(#name, &kagome::runtime::wavm::name##Intrinsic)
```

Comment:
this call don't make sense to me

Reply:
Right, it did before, when I tried to simplify intrinsic registration with lambdas, but it does not anymore. I will remove it.